pkg/kubectl/metricsutil/metrics_printer.go

```
@@ -53,16 +53,114 @@ func NewTopCmdPrinter(out io.Writer) *TopCmdPrinter {
+	switch n.sortBy {
+	case "cpu":
+		qi := n.usages[i][v1.ResourceCPU]
+		qj := n.usages[j][v1.ResourceCPU]
+		return qi.Value() > qj.Value()
+	case "memory":
+		qi := n.usages[i][v1.ResourceMemory]
+		qj := n.usages[j][v1.ResourceMemory]
+		return qi.Value() > qj.Value()
+	default:
+		return n.metrics[i].Name < n.metrics[j].Name
+	}
+}
+
+func NewNodeMetricsSorter(metrics []metricsapi.NodeMetrics, sortBy string) (*NodeMetricsSorter, error) {
+	var usages = make([]v1.ResourceList, len(metrics))
+	if len(sortBy) > 0 {
+		for i, v := range metrics {
+			if err := scheme.Scheme.Convert(&v.Usage, &usages[i], nil); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	return &NodeMetricsSorter{
+		metrics: metrics,
+		sortBy:  sortBy,
+		usages:  usages,
+	}, nil
+}
+
+type PodMetricsSorter struct {
+	metrics       []metricsapi.PodMetrics
+	sortBy        string
+	withNamespace bool
+	podMetrics    []v1.ResourceList
+}
+
+func (p *PodMetricsSorter) Len() int {
+	return len(p.metrics)
+}
+
+func (p *PodMetricsSorter) Swap(i, j int) {
+	p.metrics[i], p.metrics[j] = p.metrics[j], p.metrics[i]
+}
+
+func (p *PodMetricsSorter) Less(i, j int) bool {
+	switch p.sortBy {
+	case "cpu":
+		qi := p.podMetrics[i][v1.ResourceCPU]
+		qj := p.podMetrics[j][v1.ResourceCPU]
+		return qi.Value() > qj.Value()
+	case "memory":
+		qi := p.podMetrics[i][v1.ResourceMemory]
+		qj := p.podMetrics[j][v1.ResourceMemory]
+		return qi.Value() > qj.Value()
+	default:
+		if p.withNamespace && p.metrics[i].Namespace != p.metrics[j].Namespace {
+			return p.metrics[i].Namespace < p.metrics[j].Namespace
+		}
+		return p.metrics[i].Name < p.metrics[j].Name
+	}
+}
+
+func NewPodMetricsSorter(metrics []metricsapi.PodMetrics, printContainers bool, withNamespace bool, sortBy string) (*PodMetricsSorter, error) {
+	var podMetrics = make([]v1.ResourceList, len(metrics))
+	if len(sortBy) > 0 {
+		for i, v := range metrics {
+			podMetrics[i], _, _ = getPodMetrics(&v, printContainers)
```

Any particular reason to swallow error?